process/transactions/scrsDataToTransactions.go

```diff
@@ -91,6 +91,13 @@ func (st *scrsDataToTransactions) fillTxWithSCRsFields(tx *data.Transaction) {
 		return
 	}
 
+	if hasCrossShardPendingTransfer(tx) {
+		tx.GasUsed = tx.GasLimit
```

Comment:
L95-L97 are duplicated on L101-L103 . please move the fee computation and setting before this if so this will only contain the return

Reply:
refactored.